spec/tracer_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# typed: false
+
+describe Redcord::Tracer do
+  class FakeTracer
+    def trace(*)
+      yield
+    end
+  end
+
+  context 'when tracer is set' do
+    include Redcord::Tracer::ClassMethods
```

Comment:
i had no idea that blocks could be used to scope includes/extends. I wonder how that works under the hood.

Reply:
I think rspec generates a runtime class/module for each context block (so the classes, methods defined here wont conflict with each other)